### PR TITLE
Documentation for publishing apps to App Store and Google Play

### PR DESCRIPTION
**Pull request by ivanbuhov**
Not ready yet. Publishing to Google Play Store should be added.

NativeScript/ios-runtime#427

### CODE REVIEW

File: core-concepts/releasing-apps.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+title: Releasing NativeScript Applications to the App Store and Google Play Store
```
Comment (ikoevska) [inline core-concepts/releasing-apps.md]: I'd go with something along the lines of:

Publishing NativeScript Apps in the App Store and Google Play
Distributing NativeScript Apps via the App Store and Google Play

File: core-concepts/releasing-apps.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+description: Learn how to release NativeScript applications to the App Store and Google Play Store
```
Comment (ikoevska) [inline core-concepts/releasing-apps.md]: Learn how to make your NativeScript apps available to end users by publishing them in the App Store or Google Play.

File: core-concepts/releasing-apps.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+position: 10
+slug: releasing-apps
+---
+
+# Releasing to App Store
```
Comment (ikoevska) [inline core-concepts/releasing-apps.md]: Publishing in the App Store

File: core-concepts/releasing-apps.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+The process of releasing NativeScript app to App Store is not much different than [releasing pure native iOS application](https://developer.apple.com/library/ios/documentation/IDEs/Conceptual/AppDistributionGuide/Introduction/Introduction.html). After making sure the NativeScript app is prepared by running
```
Comment (ikoevska) [inline core-concepts/releasing-apps.md]: Publishing NativeScript apps in the App Store is similar to [releasing pure native iOS apps](link). 

-> Rework the next lines into a clean procedure with very distinctive steps. E.g.
1. Verify that the iOS native project inside your app contains your latest changes and resources by running the following command.
   
   ```
   tns prepare ios
   ```
2. Open the iOS native project in Xcode. Your native project is located at: PATH.

And so on.

File: core-concepts/releasing-apps.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+```
+tns prepare ios
+```
+
+open the underlying project at `{app-name}/platforms/ios/{app-name}.xcodeproj` in Xcode.
+
+> The underlying Xcode project is created and maintained by the NativeScript CLI. Under the hood, many CLI commands apply modifications to the project making them transparent to the NativeScript developer.
```
Comment (ikoevska) [inline core-concepts/releasing-apps.md]: Do you really need this part? If your users have come to the point of publishing to the App Store with the CLI, they should already know how the CLI works. However, if you're willing to elaborate on this part - include a See Also section in your article and point to a place where you explain how the CLI works.

File: core-concepts/releasing-apps.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+From here, you have to [configure your Xcode project for distribution](https://developer.apple.com/library/ios/documentation/IDEs/Conceptual/AppDistributionGuide/ConfiguringYourApp/ConfiguringYourApp.html), [upload it to iTunes Connect](https://developer.apple.com/library/ios/documentation/IDEs/Conceptual/AppDistributionGuide/UploadingYourApptoiTunesConnect/UploadingYourApptoiTunesConnect.html) and [submit it to App Store](https://developer.apple.com/library/ios/documentation/LanguagesUtilities/Conceptual/iTunesConnect_Guide/Chapters/SubmittingTheApp.html). The whole process is explained in details in Apple's [App Distribution Guide](https://developer.apple.com/library/ios/documentation/IDEs/Conceptual/AppDistributionGuide/Introduction/Introduction.html).
+
+# Releasing to Google Play Store
+
```
Comment (ikoevska) [inline core-concepts/releasing-apps.md]: Publishing in Google Play

and I don't see the procedure for that :)

Comment (ivanbuhov) [inline core-concepts/releasing-apps.md]: It will appear in the near feature :)

**Comment by Plamen5kov** [comment]
ping: @ikoevska

**Comment by Plamen5kov** [comment]
related to: https://github.com/NativeScript/android-runtime/issues/290

File: core-concepts/releasing-apps.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+ 5. [Submit it to App Store](https://developer.apple.com/library/ios/documentation/LanguagesUtilities/Conceptual/iTunesConnect_Guide/Chapters/SubmittingTheApp.html).
+
+# Publishing in Google Play
+
+ 1. Verify that the Android native project inside your app contains your latest changes and resources by running the following command.
+ 
+     ```
+     tns prepare android
+     ```
+ 3. Make sure you have a .keystore file to sign your app with. [How to create  a .keystore file?](http://developer.android.com/tools/publishing/app-signing.html#signing-manually)
```
Comment (ikoevska) [inline core-concepts/releasing-apps.md]:
```
Make sure that you have a `.keystore` file to sign your app with. For more information, see [How to ...
```

File: core-concepts/releasing-apps.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ 2. Build your project in release mode by running the command:
```
Comment (ikoevska) [inline core-concepts/releasing-apps.md]: the following command

File: core-concepts/releasing-apps.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+ 
+    ```
+    tns build android --release --key-store-path [path_to_your_keystore] --key-store-password [your_key_store_password] --key-store-alias [your_alias_name] --key-store-alias-password [your_alias_password] 
+    ```
+ 3. Find the release .apk file inside: `<app_name>/platforms/android/build/outputs/apk/<app_name>-release.apk` and upload it to Google Developer Console. [How to publish an android app?](http://developer.android.com/distribute/googleplay/start.html)
```
Comment (ikoevska) [inline core-concepts/releasing-apps.md]: A few things need to be done for this last step.
- It appears, you are providing the complete path to the release apk. I suggest rephrasing the step a bit - something like "Obtain the release .apk located at ..."
- Split this step. Move the "How to publish an Android app" link in a separate step and make its label "Publish your Android app".

File: core-concepts/releasing-apps.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+     tns prepare android
+     ```
+ 2. Make sure that you have a `.keystore` file to sign your app with. For more information, see [How to create  a .keystore file?](http://developer.android.com/tools/publishing/app-signing.html#signing-manually)
+ 3. Build your project in release mode by running the following command:
+ 
+    ```
+    tns build android --release --key-store-path [path_to_your_keystore] --key-store-password [your_key_store_password] --key-store-alias [your_alias_name] --key-store-alias-password [your_alias_password] 
+    ```
+ 4. Obtain the release `.apk` located at `<app_name>/platforms/android/build/outputs/apk/<app_name>-release.apk`.
+ 5. Publish your Android app by uploading the `.apk` file to the Google Developer Console. For more information, see [How to publish an android app?](http://developer.android.com/distribute/googleplay/start.html)
```
Comment (ikoevska) [inline core-concepts/releasing-apps.md]: an Android app (in the label of the link)

File: core-concepts/releasing-apps.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+ 1. Verify that the iOS native project inside your app contains your latest changes and resources by running the following command.
+ 
+     ```
+     tns prepare ios
+     ```
+ 2. Open the iOS native project in Xcode. Your native project is located at: `{app-name}/platforms/ios/{app-name}.xcodeproj`.
+ 3. [Configure the project for distribution](https://developer.apple.com/library/ios/documentation/IDEs/Conceptual/AppDistributionGuide/ConfiguringYourApp/ConfiguringYourApp.html).
+ 4. [Upload the app to iTunes Connect](https://developer.apple.com/library/ios/documentation/IDEs/Conceptual/AppDistributionGuide/UploadingYourApptoiTunesConnect/UploadingYourApptoiTunesConnect.html).
+ 5. [Submit it to App Store](https://developer.apple.com/library/ios/documentation/LanguagesUtilities/Conceptual/iTunesConnect_Guide/Chapters/SubmittingTheApp.html).
```
Comment (ikoevska) [inline core-concepts/releasing-apps.md]: the App Store

(sorry for missing that initially)

**Comment by ikoevska** [comment]
Two very minor cosmetic changes. :+1: after applying them.

**Pull request merged**